monai/data/dataset.py

```diff
@@ -852,7 +852,8 @@ def disable_share_memory_cache(self):
         Because multiprocessing ProxyList is not supported for the GPU caching, may need to explicitly diasble it.
 
         """
-        self._cache = list(self._cache)
+        if self.runtime_cache and not dist.is_initialized():
```

Comment:
(if we decide to use this condition then it'll be more robust to do: `not 
 (dist.is_available() and dist.is_initialized())`)

Reply:
that's true, thank you. in fact we have an internal variable for that, I've updated it.